### PR TITLE
Fix spelling errors in CMS.

### PR DESCRIPTION
**Pull request by jsoref**
mark CMS_R_UNKNOWN_DIGEST_ALGORITM as compat
mark CMS_R_UNSUPPORTED_RECPIENTINFO_TYPE as compat

Split from #3459

### CODE REVIEW

**Comment by jsoref** [comment]
Note: this uses the same notation I'm proposing in #3460

**Comment by dot-asm** [comment]
For "bookkeeping" purposed I would insist on making this commit specifically about CMS and not spelling of specific word[s]. This means that ghashv8-armx.pl should not be part of it.

File: include/openssl/cms.h [hunk 1]
```diff
@@ -491,14 +491,16 @@ int ERR_load_CMS_strings(void);
 # define CMS_R_TYPE_NOT_ENVELOPED_DATA                    146
 # define CMS_R_UNABLE_TO_FINALIZE_CONTEXT                 147
 # define CMS_R_UNKNOWN_CIPHER                             148
-# define CMS_R_UNKNOWN_DIGEST_ALGORIHM                    149
+# define CMS_R_UNKNOWN_DIGEST_ALGORITHM                   149
+# define CMS_R_UNKNOWN_DIGEST_ALGORITM /*compat*/ CMS_R_UNKNOWN_DIGEST_ALGORITHM
```
Comment (dot-asm) [inline include/openssl/cms.h]: `CMS_R_*` are automatically generated and question is what does mkerr.pl do to compatibility definitions.

Comment (richsalz) [inline include/openssl/cms.h]: Any compat things must go *before* the BEGIN ERROR CODES comment.  Otherwise they will be wiped out.

Comment (jsoref) [inline include/openssl/cms.h]: So just a numerical definition? Or with a comment indicating the correct spelling?

**Comment by jsoref** [comment]
@dot-asm: sure, I can split those two out. That's perfectly reasonable.

File: include/openssl/cms.h (hunk 1 above)
Comment (richsalz) [inline include/openssl/cms.h]: There can be no custom code after the BEGIN ERROR comment.

**Comment by dot-asm** [comment]
Merged. Thanks for report.

**Pull request closed without merging**